DataReader.py

```diff
@@ -0,0 +1,99 @@
+import datetime
+import csv
+import glob
+import collections
+
+from Weather import Weather
+from View import View
+
+
+class data_reader:
+
+	annual_report = {}
+	annual_max_temp = {}
+	year = ""
+
+
+	def get_max(self, max_value, to_compare):
+		
+		if max_value < to_compare:
+			return True
+		
+		return False
+
+	def get_min(self, min_value, to_compare):
+		
+		if min_value < to_compare and min_value > 0:
+			return min_value
+		
+		return to_compare
+
+	def retreive_dir(self, dir_path):
+		if not glob.glob(dir_path):
+			print 'No files were found!'
+			return
+
+		for textfile in glob.glob(dir_path + "/*txt" ):
+			weather = self.read_file(textfile)
+			if weather:
+				self.compare_data(weather)
+
+	def compare_data(self, weather):		
+
+		if not self.annual_report.get(weather.year):
+			self.annual_report[weather.year] = weather
+			return
+		
+		if weather.max_temp > self.annual_report[weather.year].max_temp:
+			self.annual_report[weather.year].date = weather.date
+			self.annual_report[weather.year].max_temp = weather.max_temp
+
+		if weather.min_temp < self.annual_report[weather.year].min_temp:
+			self.annual_report[weather.year].min_temp = weather.min_temp
+		
+		if weather.max_humidity > self.annual_report[weather.year].max_humidity:
+			self.annual_report[weather.year].max_humidity = weather.max_humidity
+		
+		if weather.min_humidity < self.annual_report[weather.year].min_humidity:
+			self.annual_report[weather.year].min_humidity = weather.min_humidity
+	
+	def generate_annual_report(self, dir_path):
+
+		self.retreive_dir(dir_path)
+		return self.annual_report
+
+	def generate_max_temp_report(self, dir_path):
+		self.retreive_dir(dir_path)
+		return self.annual_report
+
+	def read_file(self, textfile):
+		try: 
+			with open(textfile) as file:
+				file_reader = csv.DictReader(file)
+				year = textfile.split("_")[2]
+				month = textfile.split("_")[3].split(".")[0]
+				max_temp = 0
```

Comment:
These will result in error as weather data can go below zero (For all of the following too).